<h1 style = "font-family: Comic Sans MS;background-color:#red"> Introduction </h1>

[Github link to project](https://github.com/Phatdz02/programming-for-data-science/tree/main)

Student in this final project:

- 20127272 - Trần Thịnh Phát
- 19127237 - Trương Minh Nhựt

We have selected a dataset for the "Programming for Data Science" course, focusing on the health theme. Our choice reflects an intention to explore a related topic. Additionally, we anticipate observing a positive trend in the global tech industry concerning issues related to mental health.

<h1 style = "font-family: Comic Sans MS;"> Library </h1> 

Import neccessary library for this file.

In [ ]:
import sys
sys.executable

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

import warnings
warnings.simplefilter('ignore')

plt.style.use('seaborn-whitegrid')

<h1 style = "font-family: Comic Sans MS;">  Data Collection </h1> 
    
## Objectives and Ideas
As sharing before, I decided to find the dataset about the mental health problems, especially in technology field. I found this whilst going on [Kaggle.com](https://www.kaggle.com/) which is a popular website with lots of useful dataset to start working.

## Sources
The title: *[Mental Health in Tech Survey](https://www.kaggle.com/datasets/osmi/mental-health-in-tech-survey?datasetId=311)* \
**Survey on Mental Health in the Tech Workplace in 2014**

## About Dataset
This dataset is from a 2014 survey that measures attitudes towards mental health and frequency of mental health disorders in the tech workplace.

## The author
The survey comes from **OSMI**.\
*OSMI is a non-profit 501c3 corporation dedicated to "raising awareness, educating, and providing resources to support mental wellness in the tech and **open source** communities." What they do in support of this goal includes providing e-books on mental wellness in the workplace, hosting a forum on conversations on mental health, and holding talks at developer conferences about mental health in the community.* \
Details can be found at: *[About OSMI](https://osmihelp.org/index.html)* 

## License
The dataset gets the License **CC BY-SA 4.0** for:
- Share — copy and redistribute the material in any medium or format.
- Adapt — remix, transform, and build upon the material for any purpose, even commercially.

## Is it a sample, and was it properly sampled?
Yes, it is a sample. I can't find official information about the data collection method, but it appears *not* to be a random sample (the survey link is available on their website), so I can assume that it is not representative.

## How did authors collect data?
OSMI provides a survey on mental health in tech industry. This survey contains a variety of questions pertaining to the mental health of the respondents, the demographics of the respondents, and how employer views on mental health in the workplace. This survey was conducted in 2014.

<h1 style = "font-family: Comic Sans MS;">  Exploring data </h1> 

In [2]:
df = pd.read_csv('survey.csv')
df.head(5)

Timestamp  Age  Gender         Country state self_employed  \
0  2014-08-27 11:29:31   37  Female   United States    IL           NaN   
1  2014-08-27 11:29:37   44       M   United States    IN           NaN   
2  2014-08-27 11:29:44   32    Male          Canada   NaN           NaN   
3  2014-08-27 11:29:46   31    Male  United Kingdom   NaN           NaN   
4  2014-08-27 11:30:22   31    Male   United States    TX           NaN   

  family_history treatment work_interfere    no_employees  ...  \
0             No       Yes          Often            6-25  ...   
1             No        No         Rarely  More than 1000  ...   
2             No        No         Rarely            6-25  ...   
3            Yes       Yes          Often          26-100  ...   
4             No        No          Never         100-500  ...   

                leave mental_health_consequence phys_health_consequence  \
0       Somewhat easy                        No                      No   
1          Don't know                     Maybe                      No   
2  Somewhat difficult                        No                      No   
3  Somewhat difficult                       Yes                     Yes   
4          Don't know                        No                      No   

      coworkers supervisor mental_health_interview phys_health_interview  \
0  Some of them        Yes                      No                 Maybe   
1            No         No                      No                    No   
2           Yes        Yes                     Yes                   Yes   
3  Some of them         No                   Maybe                 Maybe   
4  Some of them        Yes                     Yes                   Yes   

  mental_vs_physical obs_consequence comments  
0                Yes              No      NaN  
1         Don't know              No      NaN  
2                 No              No      NaN  
3                 No             Yes      NaN  
4         Don't know              No      NaN  

[5 rows x 27 columns]

## How many rows and columns in the dataset?

In [3]:
print('\033[1m' + 'Shape of the data (rows, columns):' + '\033[0m')
print(df.shape)

Shape of the data (rows, columns):
(1259, 27)


We can see that there are 1259 rows with 27 columns.

## What is the meaning of each column?

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1259 entries, 0 to 1258
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Timestamp                  1259 non-null   object
 1   Age                        1259 non-null   int64 
 2   Gender                     1259 non-null   object
 3   Country                    1259 non-null   object
 4   state                      744 non-null    object
 5   self_employed              1241 non-null   object
 6   family_history             1259 non-null   object
 7   treatment                  1259 non-null   object
 8   work_interfere             995 non-null    object
 9   no_employees               1259 non-null   object
 10  remote_work                1259 non-null   object
 11  tech_company               1259 non-null   object
 12  benefits                   1259 non-null   object
 13  care_options               1259 non-null   object
 14  wellness

There are 27 columns in the datset. The survey covers information mainly in 5 aspects:

1.  Geographic and Demographic information of responders: e.g. Age, Gender, Country, state, family history of mental illness.

2. Basic information about workspace: e.g. self-employed or not, number of employees, remote work or not, tech company or not, work interfere when have mental health condition.

3. Supporting for mental health from Workspaces:
- benefits: Does your employer provide mental health benefits?
- care_options: Do you know the options for mental health care your employer provides?
- wellness_program: Has your employer ever discussed mental health as part of an employee wellness program?
- seek_help: Does your employer provide resources to learn more about mental health issues and how to seek help?
- anonymity: Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources?
- leave: How easy is it for you to take medical leave for a mental health condition?

4. Possible negative consequence:
- mental_health_consequence: Do you think that discussing a mental health issue with your employer would have - - negative consequences?
- phys_health_consequence: Do you think that discussing a physical health issue with your employer would have negative consequences?
- coworkers: Would you be willing to discuss a mental health issue with your coworkers?
- supervisor: Would you be willing to discuss a mental health issue with your direct supervisor(s)?
- mental_health_interview: Would you bring up a mental health issue with a potential employer in an interview?
- phys_health_interview: Would you bring up a physical health issue with a potential employer in an interview?
mental_vs_physical: Do you feel that your employer takes mental health as seriously as physical health?
- obs_consequence: Have you heard of or observed negative consequences for coworkers with mental health conditions in your workplace?

5. Whether or not seek mental disorder treatment: treatment.

## What is the meaning of each row?


In [6]:
df.sample()

Timestamp  Age  Gender         Country state self_employed  \
630  2014-08-28 02:32:11   24  Female  United Kingdom   NaN            No   

    family_history treatment work_interfere no_employees  ...       leave  \
630             No       Yes      Sometimes      100-500  ...  Don't know   

    mental_health_consequence phys_health_consequence coworkers    supervisor  \
630                     Maybe                      No        No  Some of them   

    mental_health_interview phys_health_interview mental_vs_physical  \
630                      No                 Maybe         Don't know   

    obs_consequence comments  
630              No      NaN  

[1 rows x 27 columns]

Each row presents for each respondents about their opinion with mental health problems.

## Are there duplicated rows?

In [7]:
dups_count = df.duplicated(keep='first').sum()
dups_count != 0

False

There is NO duplicated rows in the dataset.